### PR TITLE
build: ⬆️ upgrade to react-scripts 5.0.0

### PR DESCRIPTION
**Pull request by jalezi**
upgrade:
- react-scripts 4.0.3 -> 5.0.0
- react-leaflet 3.2.2 -> 3.2.4
- react-papaparse 3.17.2 -> 3.18.2

needs some selective dependency resolutions:
- postcss-normalize: 10.0.1
- mini-css-extract-plugin: 2.4.5

Run `yarn install`. In case it's not working:
- delete node_modules
- delete yarn.lock
- yarn install

### CODE REVIEW

**Comment by jalezi** [comment]
@overlordtm Maybe this was not a good idea 🥺. `prettier` went nuts 😵‍💫.

File: package.json [hunk 1]
```diff
@@ -25,22 +25,22 @@
...
+    "react-scripts": "^5.0.0",
     "react-style-proptype": "^3.2.2",
     "slugify": "^1.6.4",
     "uuid": "^8.3.2",
     "web-vitals": "^2.1.2"
   },
   "scripts": {
-    "start": "cross-env SKIP_PREFLIGHT_CHECK=true DISABLE_ESLINT_PLUGIN=true react-scripts start",
-    "build": "cross-env SKIP_PREFLIGHT_CHECK=true DISABLE_ESLINT_PLUGIN=true react-scripts build",
+    "start": "cross-env react-scripts start",
```
Comment (mihaerzen) [inline package.json:40]: cross-env is not really needed now anymore so we can simply remove it from all the commands. I would also suggest to remove the module all together if it's not used elsewhere.

**Comment by mihaerzen** [comment]
> @overlordtm Maybe this was not a good idea 🥺. `prettier` went nuts 😵‍💫. 

What exactly do you mean? Looking at the PR everything seems fine.

**Comment by jalezi** [comment]
> > @overlordtm Maybe this was not a good idea 🥺. `prettier` went nuts 😵‍💫.
> 
> What exactly do you mean? Looking at the PR everything seems fine.

build deploy failed
build-failed

**Comment by mihaerzen** [comment]
> > > @overlordtm Maybe this was not a good idea 🥺. `prettier` went nuts 😵‍💫.
> 
> > 
> 
> > What exactly do you mean? Looking at the PR everything seems fine.
> 
> 
> 
> build deploy failed
> 
> build-failed
> 
> 
> 
> 

I see. But the test action was successful. We have to make sure these two tasks aren't diverging with linting.

**Comment by mihaerzen** [comment]
I can have a look in the evening 👀

File: package.json [hunk 2]
```diff
@@ -25,22 +25,22 @@
...
+    "build": "cross-env react-scripts build",
```
Comment (mihaerzen) [inline package.json:41]: ~~cross-env is not really needed now anymore so we can simply remove it from all the commands. I would also suggest to remove the module all together if it's not used elsewhere.~~

This needs to be added back in order for build to work:

```suggestion
    "start": "cross-env DISABLE_ESLINT_PLUGIN=true react-scripts start",
    "build": "cross-env DISABLE_ESLINT_PLUGIN=true react-scripts build",
```

File: package.json [hunk 3]
```diff
@@ -25,22 +25,22 @@
...
     "lint": "eslint .",
-    "test": "cross-env SKIP_PREFLIGHT_CHECK=true DISABLE_ESLINT_PLUGIN=true react-scripts test",
+    "test": "cross-env react-scripts test",
```
Comment (mihaerzen) [inline package.json:43]:
```suggestion
    "test": "cross-env DISABLE_ESLINT_PLUGIN=true react-scripts test",
```

File: package.json (hunk 2 above)
Comment (mihaerzen) [reply package.json:41]: OR at least on the build command. It works locally for me without disabling the eslint plugin but in the docker container it fails. I can't say why.

Comment (mihaerzen) [reply package.json:41]: OK, I have found the issue. This is the line that is causing the build to fail:

https://github.com/sledilnik/zdravniki/blob/master/.dockerignore#L5

Basically `.prettierrc.json` is not copied to the docker image and therefore prettier inside of the docker fails as it has a different configuration. I would suggest removing it.

Afterwards, it is also safe to remove all the `cross-env` it seems.

Comment (mihaerzen) [reply package.json:41]: @jalezi let me know if you need anymore help.

**Comment by jalezi** [comment]
With `react-leaflet@3.2.4` and higher we get an error when user navigates to dr's page and then tries to return with "Nazaj to imenik" link.

react-leaflet@3 2 4_error.

I also removed `cross-env` package.

**Pull request merged**